tools/testbench/topology.c

```diff
@@ -724,16 +724,26 @@ int parse_topology(struct sof *sof, struct shared_lib_table *library_table,
 		switch (hdr->type) {
 		/* load dapm widget */
 		case SND_SOC_TPLG_TYPE_DAPM_WIDGET:
+		{
 			sprintf(message, "number of DAPM widgets %d\n",
 				hdr->count);
 
 			debug_print(message);
 
 			num_comps += hdr->count;
 			size = sizeof(struct comp_info) * num_comps;
-			temp_comp_list = (struct comp_info *)
+			struct comp_info *comp_list_realloc = (struct comp_info *)
```

Comment:
Typically we declare vars at top of function call (my guess is for older C version support) so lets also do that here so we don't break anything.

Reply:
Ack. Made the changes.